File: 1576-reorder-routes-to-make-all-paths-lead-to-the-city-zero/reorder-routes-to-make-all-paths-lead-to-the-city-zero.cpp
class Solution {
    public:
    int m=0;
    void dfs(int n, vector<vector<pair<int,int>>>&adj,vector<int>&vis,int node)
    {
        vis[node]=1;
        for(auto it:adj[node])
        {
            if(!vis[it.first])
            {m+=it.second;
            dfs(n,adj,vis,it.first);
            }


        }
    }
public:

    int minReorder(int n, vector<vector<int>>& connections) {
        vector<vector<pair<int,int>>>adj(n);
        for(auto it:connections)
        {
            int u=it[0];
            int v=it[1];
            adj[u].push_back({v,1});
             adj[v].push_back({u,0});
        }
            vector<int>vis(n,0);
            dfs(n,adj,vis,0);
            return abs(m);

    }
};

File: 1576-reorder-routes-to-make-all-paths-lead-to-the-city-zero/reorder-routes-to-make-all-paths-lead-to-the-city-zero.py
class Solution:
    def minReorder(self, n: int, connections: List[List[int]]) -> int:
        adj=defaultdict(list)
        for u,v in connections:
            adj[u].append((v,1))
            adj[v].append((u,0))
        sum=0
        vis=[0]*n
        def dfs(node,adj):
            nonlocal sum
            vis[node]=1
            
            for child,weight in adj[node]:
                if vis[child]==0:
                    sum+=weight
                    dfs(child,adj)
            
        dfs(0,adj)
        return sum


        

        